### ISSUE TITLE
Change password type from string

### ISSUE THREAD

**Issue by YoussefAbady** [meziantou/Meziantou.Framework#499]
As you know string data type is not secured as it's immutable and couldn't be removed using garbage collector so it could be dumped from the memory as per microsoft: 

> An instance of the [System.String](https://learn.microsoft.com/en-us/dotnet/api/system.string?view=net-7.0) class is both immutable and, when no longer needed, cannot be programmatically scheduled for garbage collection; that is, the instance is read-only after it is created, and it is not possible to predict when the instance will be deleted from computer memory. Because [System.String](https://learn.microsoft.com/en-us/dotnet/api/system.string?view=net-7.0) instances are immutable, operations that appear to modify an existing instance actually create a copy of it to manipulate. Consequently, if a [String](https://learn.microsoft.com/en-us/dotnet/api/system.string?view=net-7.0) object contains sensitive information such as a password, credit card number, or personal data, there is a risk the information could be revealed after it is used because your application cannot delete the data from computer memory.

So instead you need to use another data types like byte or [SecureString](https://learn.microsoft.com/en-us/dotnet/api/system.security.securestring?view=net-7.0)

Thanks.

**Comment by meziantou** [comment]
Can you clarify where you want to use `SecureString` in this project? Which methods do you want to change?

Note: `SecureString` is being obsoleted https://github.com/dotnet/designs/pull/147/files

**Comment by YoussefAbady** [comment]
The idea is to stop using system.string to hold the password value as it's not secure and the password could be compromised easily by dumping the memory as system.string is saved in plain text in the memory. 

SecureString was a proposal you can use another data type like byte array for example. 

So the answer for your question is to replace all system.string variables in the repo that hold password value with a more secured data type.

**Comment by thoemmi** [comment]
@YoussefAbady Have you even followed the link provided by @meziantou?

Microsoft itself recommends not to use `SecureString` anymore, see https://learn.microsoft.com/en-us/dotnet/api/system.security.securestring?view=net-7.0#remarks. 
Also read the _How secure is SecureString_ in the docs: https://learn.microsoft.com/en-us/dotnet/api/system.security.securestring?view=net-7.0#how-secure-is-securestring
